clang/lib/Basic/Targets/SPIR.h

```diff
@@ -141,8 +152,15 @@ class LLVM_LIBRARY_VISIBILITY SPIR64TargetInfo : public SPIRTargetInfo {
   SPIR64TargetInfo(const llvm::Triple &Triple, const TargetOptions &Opts)
       : SPIRTargetInfo(Triple, Opts) {
     PointerWidth = PointerAlign = 64;
-    SizeType = TargetInfo::UnsignedLong;
-    PtrDiffType = IntPtrType = TargetInfo::SignedLong;
+    SizeType = TargetInfo::UnsignedLongLong;
```

Comment:
Doesn't this break Linux stuff?  This seems like it'll mess up a bunch of types.

Reply:
Yes this needs to be fixed, thanks @erichkeane

Reply:
I think size type must be defined by the host triple, not SPIR.
So I think, we shoudn't modify SPIR target at all.